enterprise/reporting/error_extractor.go

```diff
@@ -353,39 +350,31 @@ func getErrorCodeFromStatTags(statTags map[string]string) string {
 }
 
 func (ext *ExtractorHandle) isVersionDeprecationError(errorMessage string) bool {
```

Comment:
Given the performance-sensitive nature, I would write a benchmark and test other techniques as well:

For instance, create a single compiled regex combining all the terms, and check how it compares against `strings.Contains` 

regexp.Compile("not.*active|inactive|no longer.*active|not.*supported")

depending on the number of elements in the slice, a single pre-compiled regex can be more efficient